content/reference/interceptors.adoc

```diff
@@ -9,14 +9,19 @@ Michael Nygard
 ifdef::env-github,env-browser[:outfilessuffix: .adoc]
 
 
-Interceptors are the basic building block of features in Pedestal. The
+Interceptors are the basic building blocks for Pedestal features. The
 core library provides interceptors that are generally useful for
 creating HTTP web services. Applications augment those with their own
-logic to handle.
+logic.
 
-An interceptor is a pair of unary functions. Each function is called
-with a link:context-map[context map] and must return either a context
-map or a channel that will deliver a context map.
+An interceptor is defined by a clojure map containing the keys `:name`, `:enter`, and `:leave`.
+The value for `:name` is a keyword identifier for the interceptor. The values for
```

Comment:
Also, :error.

Reply:
Technically an interceptor is an instance of the record `Interceptor`. A clojure map is just one type of _thing_ that can be _become_ an interceptor since it has been extended with the `IntoInterceptor` protocol. That being said, I think it's both intentional and useful to elide these details here as the concept is explained further as you read further on.